### DISCUSSION TITLE
Screensharing with OBS Studio (+ wlrobs) & Special Workspaces!

### DISCUSSION THREAD

**Discussion by GabrielFoxDev** [hyprwm/Hyprland#2106] [Show and tell]
Even though I have thoroughly followed every guide available for screensharing with Hyprland, I still wasn't able to share my whole screen. So here's the setup I'm currently using to work around my issues.

I'm running Arch Linux on a Ryzen 3300X & GTX 950 (proprietary drivers), 1920x1080x24 @ 60 Hz.

Environment variables & ```exec-once```s
in ```~/.config/hypr/hyprland.conf```
```
env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = WLR_NO_HARDWARE_CURSORS,1
env = MOZ_ENABLE_WAYLAND,1
env = _JAVA_AWT_WM_NONREPARENTING,1
env = SDL_VIDEODRIVER,wayland
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1
env = QT_QPA_PLATFORM,wayland
env = QT_AUTO_SCREEN_SCALE_FACTOR,1


exec-once=dbus-update-activation-environment DISPLAY I3SOCK SWAYSOCK WAYLAND_DISPLAY XDG_CURRENT_DESKTOP=Hyprland
exec-once=dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once=systemctl --user import-environment WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once=systemctl --user restart xdg-desktop-portal.service
exec-once=sleep 1 && /usr/lib/xdg-desktop-portal-hyprland &
```

Packages I'm using
```
yay -S \
hyprland-nvidia-git \
xdg-desktop-portal-hyprland-git \
obs-studio \
wlrobs \
```

How to:
- Open OBS
- Disable main window preview
- Select "Wayland output (dmabuf)" as source
- Right click the empty preview area
- Select your monitor under "Fullscreen Projector (Preview)"
- Send that new fullscreen window to a special workspace
- Go to any app and share the OBS Fullscreen Projector window

That's the gist of it. You sure can automate things a bit. :+1:

**Comment by GabrielFoxDev** [comment]
Here's a little demo https://youtu.be/Jsi1hMX0DJM

**Comment by Timicercarp2840** [comment]
bro, thnx, i was looking for 2 days how to record my screen, i just want to show my desktop configuration
